packages/core/src/generators/angular/index.ts

```diff
@@ -118,6 +117,179 @@ const BINDINGS_MAPPER: { [key: string]: string | undefined } = {
+    // TODO: proper babel transform to replace. Util for this
+
+    if (keyToUse.startsWith('on')) {
+      let event = keyToUse.replace('on', '');
+      event = event.charAt(0).toLowerCase() + event.slice(1);
+
+      if (event === 'change' && node.name === 'input' /* todo: other tags */) {
+        event = 'input';
+      }
+      // TODO: proper babel transform to replace. Util for this
+      const eventName = cusArgs[0];
+      const regexp = new RegExp(
+        '(^|\\n|\\r| |;|\\(|\\[|!)' + eventName + '(\\?\\.|\\.|\\(| |;|\\)|$)',
+        'g',
+      );
+      const replacer = '$1$event$2';
+      const finalValue = removeSurroundingBlock(code.replace(regexp, replacer));
+      return ` (${event})="${finalValue}" `;
+    } else if (keyToUse === 'class') {
+      return ` [class]="${code}" `;
+    } else if (keyToUse === 'ref') {
+      return ` #${code} `;
+    } else if (
+      (VALID_HTML_TAGS.includes(node.name.trim()) || keyToUse.includes('-')) &&
+      !blockOptions.nativeAttributes.includes(keyToUse) &&
+      !Object.values(BINDINGS_MAPPER).includes(keyToUse)
+    ) {
+      // standard html elements need the attr to satisfy the compiler in many cases: eg: svg elements and [fill]
+      return ` [attr.${keyToUse}]="${code}" `;
+    } else {
+      return processCodeBlockInTemplate(keyToUse, code);
+    }
+  };
+
+const handleNgOutletBindings = (node: MitosisNode) => {
+  let allProps = '';
+  let events = '';
+  for (const key in node.bindings) {
+    if (key.startsWith('"')) {
+      continue;
+    }
+    if (key.startsWith('$')) {
+      continue;
+    }
+    const { code, arguments: cusArgs = ['event'] } = node.bindings[key]!;
+
+    if (code.includes('?')) {
+      // TODO handle ternary
+      continue;
+    } else if (key.includes('props.')) {
+      allProps += `${key.replace('props.', '')}: ${code}, `;
+    } else if (key.startsWith('on')) {
+      let event = key.replace('on', '');
+      event = event.charAt(0).toLowerCase() + event.slice(1);
+
+      if (event === 'change' && node.name === 'input' /* todo: other tags */) {
+        event = 'input';
+      }
+      // TODO: proper babel transform to replace. Util for this
+      const eventName = cusArgs[0];
+      const regexp = new RegExp(
+        '(^|\\n|\\r| |;|\\(|\\[|!)' + eventName + '(\\?\\.|\\.|\\(| |;|\\)|$)',
+        'g',
+      );
+      const replacer = '$1$event$2';
+      const finalValue = removeSurroundingBlock(code.replace(regexp, replacer));
+      events += ` (${event})="${finalValue}" `;
+    } else if (code.includes('{')) {
+      allProps += `${key}: useObjectWrapper(${handleObjectBindings(code)}) `;
+    } else if (code.startsWith('Object.values')) {
+      let stripped = code.replace('Object.values', '');
+      allProps += `${key}: useObjectDotValues${stripped} `;
+    } else if (key.includes('-')) {
+      allProps += `'${key}': ${code}, `;
+    } else {
+      allProps += `${key}: ${code}, `;
```

Comment:
if you change `processCodeBlockInTemplate` so it only receives `code` as input:

```ts
const processCodeBlockInTemplate = (code: string) => {}
```

then we should be able to reuse it here and replace all of this code. Something roughly like:

```suggestion
    } else {
      const codeToUse = processCodeBlockInTemplate(code);
      const keyToUse = key.includes('-') ? `'${key}'` : key;
      allProps += `${keyToUse}: ${codeToUse}, `;
```

Reply:
you're right! thanks! let me make the changes